addon/components/discover-page/component.js

```diff
@@ -0,0 +1,808 @@
+ *  mixing until code is combined.
+ *
+ *  How to Use:
+ *  Pass in custom text like searchPlaceholder.  The facets property will enable you to customize the filters
+ *  on the left-hand side of the discover page. Sort options are the sort dropdown options.  The lockedParams are the
+ *  query parameters that are always locked in your application. Each query parameter must be passed in individually,
+ *  so they are reflected in the URL.  Logo and custom colors must be placed in the consuming application's stylesheet. Individual components
+ *  can additionally be overridden in your application.
+ *
+ * Sample usage:
+ * ```handlebars
+ *{{discover-page
+ *    activeFilters=activeFilters
+ *    detailRoute=detailRoute
+ *    discoverHeader=discoverHeader
+ *    facets=facets
+ *    fetchedProviders=model
+ *    filterMap=filterMap
+ *    filterReplace=filterReplace
+ *    lockedParams=lockedParams
+ *    page=page
+ *    provider=provider
+ *    q=q
+ *    queryParams=queryParams
+ *    searchPlaceholder=searchPlaceholder
+ *    showActiveFilters=showActiveFilters
+ *    sortOptions=sortOptions
+ *    subject=subject
+* }}
+ * ```
+ * @class discover-page
+ */
+
+const MAX_SOURCES = 500;
+let filterQueryParams = ['subject', 'provider', 'tags', 'sources', 'publishers', 'funders', 'institutions', 'organizations', 'language', 'contributors', 'type'];
+
+export default Ember.Component.extend(Analytics, hostAppName, {
+    layout,
+    theme: Ember.inject.service(),
+    i18n: Ember.inject.service(),
+    classNames: ['discover-page'],
+    // ************************************************************
+    // PROPERTIES
+    // ************************************************************
+    /**
+     * Primary filters for service - currently setup for PREPRINTS and REGISTRIES. Ember-SHARE's equivalent is facetStates.
+     * @property {Object} activeFilters
+     * @default { providers: [], subjects: [], types: [] }
+     */
+    activeFilters: { providers: [], subjects: [], types: [] },
+    /**
+     * Contributors query parameter.  If "contributors" is one of your query params, it must be passed to the component so it can be reflected in the URL.
+     * @property {String} contributors
+     */
+    contributors: '',
+    /**
+     * Name of detail route for consuming application, like "content" or "detail". Override if search result title should link to detail route.
+     * @property {String} detailRoute
+     */
+    detailRoute: null,
+    /**
+     * Text header for top of discover page.
+     * @property {String} discoverHeader
+     */
+    discoverHeader: null,
+    /**
+     * End query parameter.  If "end" is one of your query params, it must be passed to the component so it can be reflected in the URL.
+     * @property {String} end
+     */
+    end: '',
+    /**
+     * A list of the components to be used for the search facets.
+     * @property {Array} facets
```

Comment:
Facets is something the user passes in, which is why I added it to the YUIDOC comments.  I made the default a computed property, which is how Ember-SHARE is doing this, but user can override. This is how preprints and registries can specify which search facets are available.